// @datapos/datapos-engine-support/dist/datapos-engine-support.es.js
/**
 * @author Jonathan Terrell <terrell.jm@gmail.com>
 * @copyright 2023 Jonathan Terrell
 * @file datapos-engine-support/src/component.ts
 * @license ISC
 */
var d = /* @__PURE__ */ ((e) => (e.Connection = "connection", e.ContextModel = "contextModel", e.DataConnector = "dataConnector", e.NodeConnector = "nodeConnector", e.ResultTemplate = "resultTemplate", e.UsageKit = "usageKit", e))(d || {}), g = /* @__PURE__ */ ((e) => (e.Proposed = "proposed", e.UnderReview = "under-review", e.PreAlpha = "pre-alpha", e.Alpha = "alpha", e.Beta = "beta", e.ReleaseCandidate = "release-candidate", e.GeneralAvailability = "general-availability", e.Unavailable = "unavailable", e))(g || {});
/**
 * @author Jonathan Terrell <terrell.jm@gmail.com>
 * @copyright 2023 Jonathan Terrell
 * @file datapos-engine-support/src/connector.ts
 * @license ISC
 */
var x = /* @__PURE__ */ ((e) => (e.APIKey = "apiKey", e.Disabled = "disabled", e.OAuth2 = "oAuth2", e.None = "none", e))(x || {});
const u = {
  application: { label: "Application" },
  curatedDataset: { label: "Curated Dataset" },
  database: { label: "Database" },
  fileStore: { label: "File Store" }
}, B = (e) => u[e] ? u[e] : { label: e }, b = {
  alpha: { color: "#d62728", label: "alpha" },
  beta: { color: "#8c564b", label: "beta" },
  generalAvailability: { label: "" },
  preAlpha: { color: "#d62728", label: "pre-alpha" },
  proposed: { color: "#666666", label: "proposed" },
  releaseCandidate: { color: "#ff7f0e", label: "release-candidate" },
  unavailable: { color: "#d62728", label: "unavailable" },
  underReview: { color: "#666666", label: "under-review" }
}, A = (e) => b[e] ? b[e] : { color: "#984ea3", label: e };
var v = /* @__PURE__ */ ((e) => (e.Bidirectional = "bidirectional", e.Destination = "destination", e.Node = "node", e.Source = "source", e.None = "none", e))(v || {});
/**
 * @author Jonathan Terrell <terrell.jm@gmail.com>
 * @copyright 2023 Jonathan Terrell
 * @file datapos-engine-support/src/connectionEntry.ts
 * @license ISC
 */
var p = /* @__PURE__ */ ((e) => (e.JSON = "json", e.Table = "table", e.Uint8Array = "uint8Array", e))(p || {}), y = /* @__PURE__ */ ((e) => (e.File = "file", e.Folder = "folder", e))(y || {}), h = /* @__PURE__ */ ((e) => (e.DelimitedText = "dtv", e.EntityEvent = "e/e", e.JSON = "json", e.SPSS = "spss", e.Table = "table", e.XLS = "xls", e.XLSX = "xlsx", e.XML = "xml", e))(h || {}), r = /* @__PURE__ */ ((e) => (e.Binary = "binary", e.Boolean = "boolean", e.Date = "date", e.DateTime = "dateTime", e.DateTimeOffset = "dateTimeOffset", e.DecimalNumber = "decimalNumber", e.Object = "object", e.String = "string", e.Time = "time", e.Unknown = "unknown", e.WholeNumber = "wholeNumber", e))(r || {}), E = /* @__PURE__ */ ((e) => (e.Colon = ":", e.Comma = ",", e.ExclamationMark = "!", e.Other = "", e.RecordSeparator = "0x1E", e.Semicolon = ";", e.Space = " ", e.Tab = "	", e.Underscore = "_", e.UnitSeparator = "0x1F", e.VerticalBar = "|", e))(E || {});
/**
 * @author Jonathan Terrell <terrell.jm@gmail.com>
 * @copyright 2023 Jonathan Terrell
 * @file datapos-engine-support/src/dataConnector.ts
 * @license ISC
 */
var T = /* @__PURE__ */ ((e) => (e.Binary = "binary", e.Boolean = "boolean", e.Byte = "byte", e.Date = "date", e.DateTime = "dateTime", e.DateTimeOffset = "dateTimeOffset", e.Decimal = "decimal", e.Double = "double", e.Int8 = "int8", e.Int16 = "int16", e.Int32 = "int32", e.Int64 = "int64", e.Object = "object", e.Single = "single", e.String = "string", e.Time = "time", e.Unknown = "unknown", e))(T || {});
const F = (e, t) => {
  switch (e) {
    case "Edm.Binary":
      return { storageTypeId: "binary", usageTypeId: r.Binary };
    case "Edm.Boolean":
      return { storageTypeId: "boolean", usageTypeId: r.Boolean };
    case "Edm.Byte":
      return { storageTypeId: "byte", usageTypeId: r.WholeNumber };
    case "Edm.DateTime":
      return { storageTypeId: "dateTime", usageTypeId: r.DateTime };
    case "Edm.DateTimeOffset":
      return { storageTypeId: "dateTimeOffset", usageTypeId: r.DateTimeOffset };
    case "Edm.Decimal":
      return { storageTypeId: "decimal", usageTypeId: r.DecimalNumber };
    case "Edm.Double":
      return { storageTypeId: "double", usageTypeId: r.DecimalNumber };
    case "Edm.Guid":
      return { storageTypeId: "string", usageTypeId: r.String };
    case "Edm.Int16":
      return { storageTypeId: "int16", usageTypeId: r.WholeNumber };
    case "Edm.Int32":
      return { storageTypeId: "int32", usageTypeId: r.WholeNumber };
    case "Edm.Int64":
      return { storageTypeId: "int64", usageTypeId: r.WholeNumber };
    case "Edm.SByte":
      return { storageTypeId: "int8", usageTypeId: r.WholeNumber };
    case "Edm.Single":
      return { storageTypeId: "single", usageTypeId: r.DecimalNumber };
    case "Edm.String":
      return { maximumLength: t, storageTypeId: "string", usageTypeId: r.String };
    case "Edm.Time":
      return { storageTypeId: "time", usageTypeId: r.Time };
    default:
      return { storageTypeId: "unknown", usageTypeId: r.Unknown };
  }
};
/**
 * @author Jonathan Terrell <terrell.jm@gmail.com>
 * @copyright 2023 Jonathan Terrell
 * @file datapos-engine-support/src/nodeConnector.ts
 * @license ISC
 */
var w = /* @__PURE__ */ ((e) => (e.Dimension = "dimension", e.Entity = "entity", e.EventQuery = "eventQuery", e.SourceView = "sourceView", e.Workbook = "workbook", e))(w || {}), N = /* @__PURE__ */ ((e) => (e.Data = "data", e.Events = "events", e.Facts = "facts", e))(N || {});
/**
 * @author Jonathan Terrell <terrell.jm@gmail.com>
 * @copyright 2023 Jonathan Terrell
 * @file datapos-engine-support/src/utilities.ts
 * @license ISC
 */
const W = (e) => {
  if (e) {
    const t = e.lastIndexOf("/");
    if (t > -1)
      return e.substring(0, t);
  }
}, $ = (e) => {
  if (e) {
    const t = e.lastIndexOf(".");
    if (t > -1)
      return e.substring(t + 1);
  }
}, S = (e) => {
  if (e) {
    let t, n;
    if (e.endsWith("/") ? (t = e.lastIndexOf("/", e.length - 2), n = e.length - 1) : (t = e.lastIndexOf("/"), n = e.length), t > -1)
      return e.substring(t + 1, n);
  }
}, m = "en-US", c = {}, l = (e, t = 2, n = t, a = m) => {
  if (e == null)
    return "";
  const s = `${a}decimal${t}.${n}`;
  let i = c[s];
  return i || (i = new Intl.NumberFormat(a, {
    localeMatcher: "best fit",
    maximumFractionDigits: t,
    minimumFractionDigits: n,
    minimumIntegerDigits: 1,
    style: "decimal",
    useGrouping: !0
  }), c[s] = i), i.format(e);
}, j = (e) => e == null ? "" : e === 1 ? "1 byte" : e < 1024 ? `${k(e)} bytes` : e < 1048576 ? `${l(e / 1024, 2, 0)} KB` : e < 1073741824 ? `${l(e / 1048576, 2, 0)} MB` : e < 1099511627776 ? `${l(e / 1073741824, 2, 0)} GB` : `${l(e / 1099511627776, 2, 0)} TB`, k = (e, t = m) => {
  if (e == null)
    return "";
  const n = `${t}decimal0.0`;
  let a = c[n];
  return a || (a = new Intl.NumberFormat(t, {
    localeMatcher: "best fit",
    maximumFractionDigits: 0,
    minimumFractionDigits: 0,
    minimumIntegerDigits: 1,
    style: "decimal",
    useGrouping: !0
  }), c[n] = a), a.format(e);
}, L = (e) => {
  switch (e) {
    case "csv":
      return "text/csv";
    case "tab":
    case "tsv":
      return "text/tab-separated-values";
    case "xls":
      return "application/vnd.ms-excel";
    case "xlsx":
      return "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet";
    default:
      return "application/octet-stream";
  }
}, M = async (e, t, n, a) => {
  let s;
  if (e.authorization[""].expires_at - Date.now() < 3e5) {
    const i = {
      "Account-Id": t,
      Authorization: n,
      "Connection-Id": e.id
    }, o = await fetch(a, { headers: i });
    if (!o.ok) {
      const f = {
        body: {
          context: "establishVendorAccessToken",
          message: await o.text()
        },
        statusCode: o.status,
        statusText: o.statusText
      };
      throw new Error(
        "Unable to establish access token.|" + JSON.stringify(f)
      );
    }
    s = await o.text();
  } else
    s = e.authorization[""].access_token;
  return s;
};
/**
 * @author Jonathan Terrell <terrell.jm@gmail.com>
 * @copyright 2023 Jonathan Terrell
 * @file datapos-engine-support/src/index.ts
 * @license ISC
 */
const O = {
  dataConnectors: {},
  storageURLPrefix: ""
}, R = () => O;
export {
  g as ComponentStatusId,
  d as ComponentTypeId,
  p as ConnectionEntryPreviewTypeId,
  y as ConnectionEntryTypeId,
  x as ConnectorAuthMethodId,
  v as ConnectorUsageId,
  h as DataFormatId,
  T as DataStorageTypeId,
  r as DataUsageTypeId,
  N as NodeDataTypeId,
  w as NodeItemTypeId,
  E as ValueDelimiterId,
  F as convertODataTypeToDataType,
  M as establishVendorAccessToken,
  W as extractDirectoryPathFromEntryPath,
  $ as extractExtensionFromEntryPath,
  S as extractLastFolderNameFromFolderPath,
  l as formatNumberAsDecimalNumber,
  j as formatNumberAsStorageSize,
  k as formatNumberAsWholeNumber,
  B as lookupConnectorCategory,
  A as lookupConnectorStatus,
  L as lookupMimeTypeForFileExtension,
  R as useEngine
};
